File: Patterns/Facade Pattern/CSharp/HomeTheater/Projector.cs
using System;

namespace HomeTheater
{
    public class Projector
    {
        private string description;

        public Projector(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void WideScreenMode() => Console.WriteLine($"{description} in wide screen mode (16x9 aspect ratio)");

        public void TvMode() => Console.WriteLine($"{description} in tv mode (4x3 aspect ratio)");

        public string Description { get => description; private set => description = value; }
    }
}

File: Patterns/Facade Pattern/Cpp/HomeTheater/Projector.cpp
#pragma once

#include <iostream>
#include <string>

#include "./DvdPlayer.cpp"

class Projector {
private:
    string description;

public:
    Projector(string description) : description{description} {}

    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void wideScreenMode() {
        cout << description << " in wide screen mode (16x9 aspect ratio)\n";
    }

    void tvMode() {
        cout << description << " in tv mode (4x3 aspect ratio)\n";
    }

    string toString() {
        return description;
    }
};